### DISCUSSION TITLE
3 layers data tiering with minio

### DISCUSSION THREAD

**Discussion by cubbasa** [minio/minio#15700] [Q&A] [answered]
I'm looking for solution regarding 3 layers of cascade data tiering based on prefixes and elapsed days with following flow:
`hot(minio) -> warm (minio with 30 days) -> cold (azure/aws with 90 days or more)`

With AWS s3 is possible to have overlapping lifecycle configuration with different storage classes and elapsed days so even warm and cold have lifecycle configuration that overlap each other AWS s3 will pick the cheapest one and move data there (https://docs.aws.amazon.com/AmazonS3/latest/userguide/lifecycle-configuration-examples.html#lifecycle-config-conceptual-ex5).
I'm looking for something similar with minio that I could have 3 layers of data tiering that overlap and minio pick proper one.

I've got working data tiering with 2 layers based on prefixes and elapsed days:
`hot (minio) -> warm (minio with 30 days)`

In documentation and github related to minio I wasn't able to find anything besides 2 layers so I'm not sure if 3 layers are supported at all or by using them it could lead to for example to some race conditions between different rules in lifecycle policy and/or get some unpredicted behavior that could lead to issues with data integrity?

**Comment by harshavardhana** [comment, marked as answer]
We would love to learn more about your use-case please reach out to us at https://min.io/pricing @cubbasa

**Comment by shaneday** [comment]
Did you find a solution? I'm facing a similar requirement.

In my case, I want hot data in MinIO using fast local NVMe (~$100/TB), warm data moved to Linode S3 Object-Storage ($20/TB), and cold data moved to Wasabi S3 ($6/TB).
